Flutter_Weather/lib/widgets/weather_day_detail_widget.dart

```diff
@@ -15,41 +18,49 @@ class WeatherDayDetail extends StatefulWidget {
 class _WeatherDayDetailState extends State<WeatherDayDetail> {
   @override
   Widget build(BuildContext context) {
-    final _mockWeatherData = widget.mockWeatherData;
-    final DayWeather todayData = _mockWeatherData.today;
-    final initTime = _mockWeatherData.initDate.hour;
-    return Column(
-      children: [
-        Row(
-          crossAxisAlignment: CrossAxisAlignment.end,
-          children: [
-            ..._weatherInfoDay(todayData, initTime)
-          ],
-        )
-      ],
-    );
+    final Weather _mockWeatherData = widget.mockWeatherData;
+    final DateTime _initDate = _mockWeatherData.initDate;
+    final List<DayWeather> _allAvailableDays = _mockWeatherData.allAvailableDays();
+    return Padding(
+        padding: EdgeInsets.fromLTRB(0,0,0, 0),
+        child: SizedBox(
+          height: 300,
+          child: ListView.builder(
+            itemCount: _allAvailableDays.length,
+            itemBuilder: (context, index){
+              DayWeather dayWeather = _allAvailableDays[index];
+              return
+                WeatherDayDetailList(weatherInfoDay: _weatherInfoDay(dayWeather, _initDate), weekDay: dayWeather.weekDay,);
+            },
+            scrollDirection: Axis.horizontal,
+
+      )
+    ));
   }
 
   WeatherTypeIcon _weatherIcon(String weather) {
     return WeatherTypeIcon(weather: weather);
   }
 
-  List<Widget> _weatherInfoDay(DayWeather dayData, int initTime) {
+  List<Widget> _weatherInfoDay(DayWeather dayData, DateTime initDate) {
     List<Widget> _weatherInfoDay = [];
-    for (int i = 0; i < 10; i++) {
+    for (int i = 1; i < dayData.weathers.length*2; i++) {
```

Comment:
Use ListView.separated instead, you don't have to use for loop with double of items.